### PR TITLE
Change the way the miner executable reports execution errors (closes issue #1677)

### PR DESCRIPTION
**Pull request by Aleoami**

### CODE REVIEW

File: cmd/kaspaminer/main.go [hunk 1]
```diff
@@ -38,20 +35,20 @@ func main() {
 
 	client, err := newMinerClient(cfg)
 	if err != nil {
-		panic(errors.Wrap(err, "error connecting to the RPC server"))
+		printErrorAndExit(fmt.Errorf("Error connecting to the RPC server: %s", err))
```
Comment (svarogg) [inline cmd/kaspaminer/main.go:38]: Please use errors.Errorf instead (in all cases)

**Review by svarogg — CHANGES_REQUESTED** [review]

**Review by Aleoami — COMMENTED** [review]
Done

**Review by Aleoami — COMMENTED** [review]
fixed import

**Review by svarogg — APPROVED** [review]

File: cmd/kaspaminer/main.go [hunk 2]
```diff
@@ -38,20 +37,20 @@ func main() {
 
 	client, err := newMinerClient(cfg)
 	if err != nil {
-		panic(errors.Wrap(err, "error connecting to the RPC server"))
+		printErrorAndExit(errors.Errorf("Error connecting to the RPC server: %s", err))
```
Comment (someone235) [inline cmd/kaspaminer/main.go:40]: This will lose the error stack information

File: cmd/kaspaminer/main.go [hunk 3]
```diff
@@ -38,20 +37,20 @@ func main() {
...
-		panic(errors.Wrap(err, "error decoding mining address"))
+		printErrorAndExit(errors.Errorf("Error decoding mining address: %s", err))
 	}
 
 	doneChan := make(chan struct{})
 	spawn("mineLoop", func() {
 		err = mineLoop(client, cfg.NumberOfBlocks, *cfg.TargetBlocksPerSecond, cfg.MineWhenNotSynced, miningAddr)
 		if err != nil {
-			panic(errors.Wrap(err, "error in mine loop"))
+			printErrorAndExit(errors.Errorf("Error in mine loop: %s", err))
```
Comment (someone235) [inline cmd/kaspaminer/main.go:53]: This will lose the error stack information

**Review by someone235 — CHANGES_REQUESTED** [review]

**Pull request merged**